ksql-engine/src/main/java/io/confluent/ksql/function/udaf/topk/TopkKudaf.java

```diff
@@ -24,34 +24,39 @@
 import org.apache.kafka.connect.data.Schema;
 import org.apache.kafka.connect.data.Struct;
 import org.apache.kafka.streams.kstream.Merger;
+import org.apache.kafka.streams.kstream.ValueMapper;
 
 public class TopkKudaf<T extends Comparable<? super T>>
-    extends BaseAggregateFunction<T, List<T>> {
+    extends BaseAggregateFunction<T, List<T>, List<T>> {
 
   private final int topKSize;
   private final Class<T> clazz;
-  private final Schema returnType;
+  private final Schema aggregateType;
```

Comment:
These are always the same right? Just keep one. That'll reduce the number of args in your constructor too.